client/client.go

```diff
@@ -185,3 +185,8 @@ func (c *Client) GetPaymentChannel(id types.Destination) (query.PaymentChannelIn
 func (c *Client) GetLedgerChannel(id types.Destination) (query.LedgerChannelInfo, error) {
 	return query.GetLedgerChannelInfo(id, c.store)
 }
+
+// Stop stops the client's engine run loop
+func (c *Client) Stop() {
+	c.engine.Stop()
```

Comment:
Do we need to stop/cleanup  the handleEngineEvents [goroutine](https://github.com/statechannels/go-nitro/blob/95f9fd1e702310901e38cceea911483f9305ca3a/client/client.go#L68)?